core/rsyslog/config.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package rsyslog
+
+import (
+	"crypto/tls"
+	"net/url"
+
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/cert"
+)
+
+var ErrNotConfigured = errors.Errorf("rsyslog forwarding not configured")
+
+func IsNotConfigured(err error) bool {
+	return errors.Cause(err) == ErrNotConfigured
+}
+
+// ClientConfig holds information needed to connect to a
+// rsyslog server.
+type ClientConfig struct {
+	URL    string
+	CACert string
+	Cert   string
+	Key    string
+}
+
+// Validate validates the config values.
+func (c *ClientConfig) Validate() error {
+	if c.URL == "" || c.CACert == "" || c.Cert == "" || c.Key == "" {
```

Comment:
hm, was expecting ErrNotConfigured to be equivalent to IsZero. I *think* that any non-empty field indicates an attempt to configure and pushes us into NotValid territory